qadence/backends/pyqtorch/convert_ops.py

```diff
@@ -281,7 +301,8 @@ def __init__(
                 use_full_support=False,
             )
             hmat = hmat.permute(1, 2, 0)
-            self._hamiltonian = lambda x: hmat
+            self.register_buffer("hmat", hmat)
+            self._hamiltonian = lambda self, values: self.hmat
```

Comment:
Not quite sure why this has to be a lambda ?

Reply:
yes, we need `self` since the state (for example the device) can change after instantiation.